### DISCUSSION TITLE
[Bug] bypassPullRequestAllowances empty for private repos

### DISCUSSION THREAD

**Discussion by electricgull** [community/community#36429] [Apps, API and Webhooks]
The bypassPullRequestAllowances is empty when querying a private repo in an enterprise org. When I query against a public repo, bypassPullRequestAllowances returns as expected. 

{
  repository(name: "some_private_repo", owner: "some_org") {
    name
    branchProtectionRules(first: 20) {
      nodes {
        requiresLinearHistory
        requiredApprovingReviewCount
        dismissesStaleReviews
        requiresCodeOwnerReviews
        requiresCommitSignatures
        isAdminEnforced
        allowsForcePushes
        allowsDeletions
        blocksCreations
        requiresConversationResolution
        pattern
        allowsDeletions
        requiresApprovingReviews
        requiredStatusCheckContexts
        creator {
          login
        }
        databaseId
        requiresStatusChecks
        requiresStrictStatusChecks
        restrictsPushes
        restrictsReviewDismissals
        requiredStatusChecks {
          context
          app {
            id
            name
          }
        }
        bypassPullRequestAllowances(first: 20) {
          nodes {
            actor {
              __typename
              ... on App {
                id
                name
              }
              ... on Team {
                id
                name
              }
              ... on User {
                id
                login
                email
              }
            }
          }
        }
      }
    }
  }
}

return:
{
  "data": {
    "repository": {
      "name": "some_private_repo",
      "branchProtectionRules": {
        "nodes": [
          {
            "requiresLinearHistory": false,
            "requiredApprovingReviewCount": null,
            "dismissesStaleReviews": false,
            "requiresCodeOwnerReviews": false,
            "requiresCommitSignatures": false,
            "isAdminEnforced": true,
            "allowsForcePushes": false,
            "allowsDeletions": false,
            "blocksCreations": true,
            "requiresConversationResolution": false,
            "pattern": "main",
            "requiresApprovingReviews": false,
            "requiredStatusCheckContexts": [],
            "creator": {
              "login": "some_login"
            },
            "databaseId": 11111111,
            "requiresStatusChecks": true,
            "requiresStrictStatusChecks": true,
            "restrictsPushes": true,
            "restrictsReviewDismissals": false,
            "requiredStatusChecks": [],
            "bypassPullRequestAllowances": {
              "nodes": []
            }
          }
        ]
      }
    }
  }
}

Same query against public repo
{
  "data": {
    "repository": {
      "name": "some_public_repo",
      "branchProtectionRules": {
        "nodes": [
          {
            "requiresLinearHistory": false,
            "requiredApprovingReviewCount": 1,
            "dismissesStaleReviews": true,
            "requiresCodeOwnerReviews": true,
            "requiresCommitSignatures": false,
            "isAdminEnforced": false,
            "allowsForcePushes": false,
            "allowsDeletions": false,
            "blocksCreations": false,
            "requiresConversationResolution": false,
            "pattern": "main",
            "requiresApprovingReviews": true,
            "requiredStatusCheckContexts": [],
            "creator": {
              "login": "some_login"
            },
            "databaseId": 111111222,
            "requiresStatusChecks": false,
            "requiresStrictStatusChecks": true,
            "restrictsPushes": false,
            "restrictsReviewDismissals": false,
            "requiredStatusChecks": [],
            "bypassPullRequestAllowances": {
              "nodes": [
                {
                  "actor": {
                    "__typename": "User",
                    "id": "U_zzzzzzLVJg",
                    "login": "some_user",
                    "email": ""
                  }
                },
                {
                  "actor": {
                    "__typename": "App",
                    "id": "A_zzzzzAqlAcM4AA4u8",
                    "name": "some_app"
                  }
                }
              ]
            }
          }
        ]
      }
    }
  }
}

**Comment by ossanna16** [comment]
Hey there @electricgull and thank you for asking a good question! I'm glad you found our community 🙂

**Comment by electricgull** [comment]
The required permissions for the above call are - repo, read:org and read:discussion. I was missing the "discussion" permission.
